src/fparser/two/tests/fortran2003/test_primary_r701.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2019 Science and Technology Facilities Council
+
+# All rights reserved.
+
+# Modifications made as part of the fparser project are distributed
+# under the following license:
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Test Fortran 2003 rule R701 : primary type.
+
+'''
+import pytest
+
+import fparser.two.Fortran2003 as f2003
+import fparser.two.utils
+
+
+# A list of [Expression, Expected Type, ToString] for the test cases.
+cases = [
```

Comment:
I expect pylint/pytest will complain about the cases variable being lower case. Note, test files need to be pylint and pycodestyle compliant (with some relaxation for pylint where it makes sense to break the rules).

Reply:
Would you like me to setup an automated checker based on https://stickler-ci.com/ for this? It would save human reviewers having to notice pylint... (and can be ignored as desired on a per PR basis)

Reply:
Thanks for the suggestion, I'll discuss this at the telco tomorrow. If it is good enough to deal with python 2 and 3, with pylint and pycodestyle, and reports back with an issue in a similar way to codecov then that would be great. In truth we would really like the checks to only be done on code changes rather than the whole code which I presume it does not do (but that is certainly not a deal breaker). A colleague hacked something together to do this for us but it was a bit flaky so we've not used it.

Reply:
Locally I've used bespoke git commands and pipes to get something similar, but [git lint](https://github.com/sk-/git-lint) looks quite appropriate.

I double checked Stickler and it currently only supports pycodestyle (the tool formerly known as pep8) and not flake8 - but it *does* only check changed lines. I'll put forward a PR for it in the future.